### PR TITLE
LPS-156915 Create a sample for FDS usage as a controlled React component

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-156915 Create a sample for FDS usage as a controlled React component](https://issues.liferay.com/browse/LPS-156915)
- Related to https://github.com/liferay-frontend/liferay-portal/pull/2325

### What is the goal of this PR?

We are providing a limited support for FDS usage as a controlled component, and adding a sample usage.

The support is limited to showing FDS items. All other FDS features that affect items are not yet supported: search, pagination,  sort, filtering etc. We plan to provide full support in a dedicated epic.

Notes:
- As a part of support solution, we are simplifying, fixing and improving performance of the view component loading. We are now asynchronously loading view component only if it is defined through `contentRendererModuleURL`. In the case of a normal view, which should be the most common cases, we are immediately setting it with `getViewComponent`. This approach removes an unnecessary render, and fixes the issue in https://github.com/liferay-frontend/liferay-portal/pull/2325. Notice no spinner in gif below. The only delay in rendering is inherent to `<react:component>`; in a pure React frontend there would be no delay. 
- In the sample, to showcase external control, we have a button that modifies `items`. It adds an item (user) on click, setting it based on the last item (see gif)
- In the sample, we need to disable management toolbar. No features in it currently work.
- Simplification in `Table.js` is unrelated to the rest of PR, I just noticed that we have an unused layer of loading logic.
- Change to usage of `ClayLoadingIndicator` is unrelated to the rest of PR, it's just a minor standardization improvement.

@Esther-OC  @john-co We can add this sample case to FDS CI tests.

/cc @dsanz @rebsilva @miltonmc 

### What does it look like?

after

### Steps to reproduce

Place `frontend-data-set-sample-web` as a widget on any page.

### CODE REVIEW

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:frontend

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js [hunk 1]
```diff
@@ -157,6 +158,8 @@ const FrontendDataSet = ({
 			}
 		}
 
+		initialActiveView.component = getViewComponent(initialActiveView.name);
+
 		return {
 			activeView: initialActiveView,
```
Comment (diegonvs) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js:164]: Mutating objects seems weird in React code; I would avoid it.

```suggestion
			activeView: {
			    ...initialActiveView,
			    component: getViewComponent(initialActiveView.name)
			},
```

Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js:164]: Yeah, that's much better! :+1:

**Comment by markocikos** [comment]
Force-pushed! Changes are in the added commit:
- We are not mutating props, as suggested in https://github.com/liferay-frontend/liferay-portal/pull/2432#discussion_r911481902
- Slight change to suggestion in the comment, I set `...initialActiveView` after `component`. That way, it's possible to override component through props.
- I reverted to using `view.contentRenderer` to determine component. Change to using `name` was a mistake. This prop exists so we can have multiple `table` views, where `name` sets them apart. This is currently irrelevant, as all usages have only a single `table` view, but will become important with the upcoming custom views.

**Comment by markocikos** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:forward

**Pull request closed without merging**